In [2]:
import pandas as pd 
import numpy as np 
from sklearn.preprocessing import StandardScaler 
from sklearn.decomposition import PCA 
from sklearn.linear_model import LinearRegression 
import matplotlib.pyplot as plt 
import seaborn as sns 

In [7]:
df = pd.read_csv("final_learning_potential_dataset.csv")
distress_vars = df[["stress_congeneric_score", "depression_congeneric_score", "cog_affect_congeneric"]]
somatic = df[["somatic_congeneric"]]

In [9]:
scaler = StandardScaler()
distress_scaled = scaler.fit_transform(distress_vars) 

In [10]:
pca = PCA(n_components=1)
latent_distress = pca.fit_transform(distress_scaled)

print("Explained Variance Ratio:", pca.explained_variance_ratio_)
print("Loadings:", pca.components_)

Explained Variance Ratio: [0.86723446]
Loadings: [[0.53710311 0.59760796 0.59530241]]


## Amaç
 Stres, depresyon ve bilişsel duygusal belirtileri temsil eden alt ölçeklerin ortak varyansını açıklayan tek bir latent "distress" faktörü elde etmek amacıyla Temel Bileşen Analizi (PCA) uygulanmıştır. 

## Yöntem 
 PCA, n_components=1 olacak şekilde yürütülmüştür. 
 Analiz öndesi değişkenler, StandardScale ile standartlaştırılmıştır. 

## Bulgular 
 PCA ÖLÇÜTÜ                      |  DEĞER 
 Açıklanan Varyans Oranı (EVR)   |  0.867
 Bileşen Yükleri                 |  0.537, 0.597, 0.595

## Yorum 
 Tek bileşen, distress alt ölçeklerindeki ortak varyansın %86.7'sini açıkalamaktadır. 
 Üç değişken de bileşene çok yakın boyutlarda yüklenmiştir. 
 Bileşen yüklerinin dengesi, latent yapının tek boyutlu ve tutarlı olduğunu göstermektedir. 
 Bu sonuç, distress yapısının ölçülen tüm alt boyutlarda ortak bir faktör tarafından temsil
 edildiğini desteklemektedir. 
